addons/web/static/src/js/views/search_panel.js

```diff
@@ -1,788 +1,221 @@
-        const newDomain = JSON.stringify([...params.searchDomain, ...params.viewDomain]);
-        if (this.reloadOptions || (currentDomain !== newDomain)) {
-            this.searchDomain = params.searchDomain;
-            this.viewDomain = params.viewDomain;
-            if (!this.reloadOptions || this.reloadOptions.shouldFetchCategories) {
-                await this._fetchCategories();
-            }
-            await this._fetchFilters();
-            this.reloadOptions = null;
-        }
-        return this._render();
-    },
-
-    //-------------------------------------------------------------------------
-    // Private
-    //-------------------------------------------------------------------------
 
-    /**
-     * Set active values for each filter (coming from context). This needs to be
-     * done only once, at widget startup.
-     *
-     * @private
-     */
-    _applyDefaultFilterValues() {
-        Object.values(this.filters).forEach(filter => {
-            const defaultValues = this.defaultValues[filter.fieldName] || [];
-            defaultValues.forEach(value => {
-                if (filter.values[value]) {
-                    filter.values[value].checked = true;
-                }
-            });
-            Object.values(filter.groups || []).forEach(this._updateFilterGroupState);
-        });
-    },
-    /**
-     * @private
-     * @param {string} categoryId
-     * @param {Object} result
-     */
-    _createCategoryTree(categoryId, result) {
-        const category = this.categories[categoryId];
-        let { error_msg, parent_field: parentField, values } = result;
-        if (error_msg) {
-            category.errorMsg = error_msg;
-            values = [];
-        }
-        if (category.hierarchize) {
-            category.parentField = parentField;
+            this._expandParentCategories();
         }
 
-        const unfoldedIds = Object.values(category.values || {})
-            .filter(c => c.folded === false)
-            .map(c => c.id);
-
-        category.values = {};
-        values.forEach(value => {
-            category.values[value.id] = Object.assign({}, value, {
-                childrenIds: [],
-                folded: !unfoldedIds.includes(value.id),
-                parentId: value[parentField] || false,
-            });
-        });
-        values.forEach(value => {
-            const { parentId: parentCategoryId } = category.values[value.id];
-            if (parentCategoryId && parentCategoryId in category.values) {
-                category.values[parentCategoryId].childrenIds.push(value.id);
-            }
-        });
-        category.rootIds = [];
-        for (const value of values) {
-            const { parentId } = category.values[value.id];
-            if (!parentId) {
-                category.rootIds.push(value.id);
+        mounted() {
+            this._updateGroupHeadersChecked();
```

Comment:
in my mind, that function should not be called here. Normally the 'check status' should be clear at this time

Reply:
The `searchpanel_default_` values in the context might check some of the filter values -> we want the group header to react accordingly on init